app/scripts/emissions-factors-sync.ts

```diff
@@ -0,0 +1,308 @@
+    region: ef.region || "global",
+    source: ef.source,
+    reference: ef.reference,
+    description: ef.description,
+    temporalGranularity: ef.temporal_granularity,
+    url: ef.url,
+    methodologyId: ef.methodology_id,
+    metadata: ef.metadata ? (typeof ef.metadata === 'string' ? JSON.parse(ef.metadata) : ef.metadata) : null,
+    deprecated: false,
+  }));
+
+  const dataSourceEmissionsFactors: DataSourceEmissionsFactorCreationAttributes[] = apiData.emissions_factor_datasource_mapping.map((mapping: EmissionsFactorDataSourceMapping) => ({
+    datasourceId: mapping.datasource_id,
+    emissionsFactorId: mapping.emissions_factor_id,
+  }));
+
+  return {
+    publishers,
+    dataSources,
+    methodologies,
+    emissionsFactors,
+    dataSourceEmissionsFactors
+  };
+}
+
+async function markExistingEmissionsFactorsAsDeprecated() {
+  logger.debug("Marking existing emissions factors as deprecated...");
+  await db.models.EmissionsFactor.update(
+    { deprecated: true },
+    { where: {} }
+  );
+}
+
+async function syncPublishers(publishers: PublisherCreationAttributes[]) {
+  logger.debug("Syncing publishers...");
+  for (const publisher of publishers) {
+    await db.models.Publisher.upsert(publisher);
+  }
+  console.log(`Synced ${publishers.length} publishers`);
+}
+
+async function syncDataSources(dataSources: DataSourceI18nCreationAttributes[]) {
+  logger.debug("Syncing data sources...");
+  for (const dataSource of dataSources) {
+    await db.models.DataSource.upsert(dataSource);
+  }
+  console.log(`Synced ${dataSources.length} data sources`);
+}
+
+async function syncMethodologies(methodologies: MethodologyCreationAttributes[]) {
+  logger.debug("Syncing methodologies...");
+  for (const methodology of methodologies) {
+    await db.models.Methodology.upsert(methodology);
+  }
+  console.log(`Synced ${methodologies.length} methodologies`);
+}
+
+async function syncEmissionsFactorsData(emissionsFactors: EmissionsFactorCreationAttributes[]) {
+  logger.debug("Syncing emissions factors...");
+  for (const emissionsFactor of emissionsFactors) {
+    await db.models.EmissionsFactor.upsert(emissionsFactor);
+  }
+  console.log(`Synced ${emissionsFactors.length} emissions factors`);
+}
+
+async function syncDataSourceEmissionsFactorRelationships(relationships: DataSourceEmissionsFactorCreationAttributes[]) {
+  logger.debug("Syncing datasource-emissionsfactor relationships...");
+  await db.models.DataSourceEmissionsFactor.bulkCreate(relationships, {
+    ignoreDuplicates: true
+  });
+  console.log(`Synced ${relationships.length} datasource-emissionsfactor relationships`);
+}
+
+async function deleteUnusedDeprecatedEmissionsFactors() {
+  logger.debug("Deleting unused deprecated emissions factors...");
+  await db.sequelize!.query(`
+    DELETE FROM "EmissionsFactor" ef
+    WHERE ef.deprecated = true AND ef.inventory_id IS NULL AND NOT EXISTS (
+      SELECT FROM "GasValue" gv WHERE gv.emissions_factor_id = ef.id
+    );
```

Comment:
I think this query isn't too complex to be expressed with Sequelize's methods. Why do we need the raw query here?